scipy/stats/_multivariate.py

```diff
@@ -5057,6 +5059,168 @@ def rvs(self, size=1, random_state=None):
+        out += np.sum(loggamma(x+A)-(loggamma(A)+loggamma(np.add(x,1))),A.ndim-1)
+
+        return out
+
+    def pmf(self, alpha, x, n):
+        """
+        Probability mass function for a Dirichlet multinomial distribution.
+        .. note::
+        The results for this function can be inaccurate when n,
+        the sum of alpha, or the sum of x is very large. If using large
+        values, consider using logpmf and exponentiating instead.
+
+        Parameters
+        ----------
+        alpha: ndarray
+            The concentration parameters. The number of
+            entries determines the dimensionality of the distribution.
+        n: int
+            Number of trials.
+
+        Returns
+        -------
+        out: scalar
+            Probability mass function for a Dirichlet multinomial distribution.
+        """
+        if alpha.shape != x.shape:
+            raise ValueError("`x` and `alpha` must have the same shape.")
+
+        A = alpha
+
+        out = gamma(np.sum(A,A.ndim-1))*gamma(n+1)/gamma(n+np.sum(A,A.ndim-1))
+        out *= np.prod(gamma(x + A) / (gamma(A) * gamma(np.add(x, 1))))
+
+        return out
+
+    def mean(self, alpha, n):
+        """
+        Mean of a Dirichlet multinomial distribution.
+
+        Parameters
+        ----------
+        alpha: ndarray
+            The concentration parameters. The number of
+            entries determines the dimensionality of the distribution.
+        n: int
+            Number of trials.
+
+        Returns
+        -------
+        out: scalar
+            Mean of a Dirichlet multinomial distribution.
+        """
+        A = alpha
+        n = np.asarray(n)
+
+        out = ((n * A.T) / (np.sum(A, A.ndim - 1))).T
+        return out
+
+    def var(self, alpha, n):
+        """
+        The variance of the Dirichlet multinomial distribution.
+
+        Parameters
+        ----------
+        %(_dirichlet_doc_default_callparams)s
+        n: int
+            Number of trials.
+
+        Returns
+        -------
+        out: array_like
+            The variances of the components of the distribution.  This is
+            the diagonal of the covariance matrix of the distribution.
+        """
+        alpha = _dirichlet_check_parameters(alpha)
+        A = alpha
+
+        out = (n*A/sum(A))*(1-(A/sum(A)))*((n+sum(A))/(1+sum(A)))
+        return out
+
+    def cov(self, alpha, n):
+        """
+        Covariance matrix of a Dirichlet multinomial distribution.
+
+        Parameters
+        ----------
+        %(_dirichlet_doc_default_callparams)s
+        n: int
+            Number of trials.
+
+        Returns
+        -------
+        out : array_like
+            The covariance matrix of the distribution
+        """
+        alpha = _dirichlet_check_parameters(alpha)
+        A = alpha
+
+        out = -n*np.outer(A,A)/(sum(A)**2)*((n+sum(A))/(1+sum(A)))
+        np.fill_diagonal(out,(n*A/sum(A))*(1-(A/sum(A)))*((n+sum(A))/(1+sum(A))))
```

Comment:
I believe the previous suggestion of saving sum(alpha) in a variable instead of recomputing at each step is still very valid here.

Reply:
Done